pyproject.toml

```diff
@@ -27,6 +27,7 @@ select = [
   "TID",    # flake8-tidy-imports
   "W",      # pycodestyle
   "YTT",    # flake8-2020
+  "UP"
```

Comment:
Code complexity of 100 below seems quite high compared to a default of 10.  It would be useful on a separate PR to add strong tests around the offending code and then lower the complexity somewhat.